---
output: html_document
---
<style type="text/css">
.main-container {
  max-width: 1800px;
  margin-left: auto;
  margin-right: auto;
}
</style>

<style>
div.gray { background-color:#7e8687; border-radius: 5px; padding: 20px;}
</style>
<div class = "gray">



<style>
div.blue { background-color:#9dc9d1; border-radius: 5px; padding: 20px;}
</style>
<div class = "blue">

## [HOME](http://talia-backman.github.io/) | [ABOUT ME](http://talia-backman.github.io/about_me) | [DATA COURSE](http://talia-backman.github.io/data_course) | [FINAL PROJECT](http://talia-backman.github.io/final_project) | [RESEARCH](http://talia-backman.github.io/research)

### **Purpose of this Rmd**
##### The purpose of the R markdown is to show the workflow of my final projet for my Data Analysis class.  My github repository has the data I used and the proper file paths. All file paths are starting from the Final_Project folder.
#### [My Github Final Project Repository](https://github.com/talia-backman/Data_Course_BACKMAN/tree/master/Assignments/Final_Project)
##### All commands were done in the command line, unless indicated otherwise.  All commands were executed using these 10 taxa:
##### * EP number is the identifier, then the Family, genus, and species are listed- some species are unknown.
- EP023: Ameletidae	Ameletus
- EP025: Siphlonuridae	Parameletus	columbiae
- EP080: Ameletopsidae	Chaquihua
- EP084: Siphlonuridae	Siphlonurus
- EP109: Siphlaenigmatidae Siphlaenigma janae
- EP150: Dipteromimidae Dipteromimus	
- EP411: Siphlonuridae	Siphlonurus	croaticus
- EP416: Siphlonuridae	Siphlonurus	lacustris
- EP418: Siphlonuridae	Siphlonurus	aestivalis
- EP429: Siphlonuridae	Siphlonisca	aerodromia.

##### This workflow only shows the commands done for EP429: Siphlonuridae	Siphlonisca	aerodromia.

___

### **Introduction**
##### Siphlonuridae is a family of mayflies (order Ephemeroptera) that are nicknamed the Primitive Minnow Mayfly family.  The family consists of four genera: Edmundsius (Day, 1953), Parameletus (Bengtsson, 1908), Siphlonisca (Needham, 1909), and Siphlonurus (Eaton, 1868).  Within the four genera are twenty six described species.  While past molecular evidence suggests that Siphlonuridae is monophyletic and includes Dipteromimidae (Ogden & Whiting, 2005), there is also research suggesting that Siphlonuridae is the sister group to Dipteromimidae (Tojo & Matsukawa, 2003).  Determining monophyly of the two taxa would help clarify and give a better understanding to the overall phylogeny of mayflies.  The conflicting arguments are both backed by molecular data, so this study aims to use metagenomic NGS data in order to generate more conclusive evidence supporting either monophyly or paraphyly.
___ 
## **Analyizing Metagenomic NGS Using the Command Line**
### **1. Quality Control using Trimmomatic**
- These parameters were used for quality control:
- LEADING:3- Cut bases off the start of a read, if below a threshold quality
- TRAILING:3- Cut bases off the end of a read, if below a threshold quality
- SLIDINGWINDOW:4:20- Performs a sliding window trimming approach. It starts scanning at the 5â€Ÿ end and clips the read once the average quality within the window falls below a threshold. 
- MINLEN:50- Drop the read if it is below a specified length
- TOPHRED33- Convert quality scores to Phred-33
```{bash, eval=FALSE}
java -jar ./Trimmomatic-0.39/trimmomatic-0.39.jar PE -threads 6 ./Taxa/EP429/EP429_R1_001.fastq ./Taxa/EP429/EP429_R2_001.fastq ./Taxa/EP429/EP429_R1_trimmed.fastq ./Taxa/EP429/EP429_R1un_trimmed.fastq ./Taxa/EP429/EP429_R2_trimmed.fastq ./Taxa/EP429/EP429_R2un_trimmed.fastq LEADING:3 TRAILING:3 SLIDINGWINDOW:4:20 MINLEN:50 TOPHRED33
```
#### This command created 4 new files (on top of the 2 raw files).  Now all the files we have are:
- EP429_R1_001.fastq 
- EP429_R2_001.fastq
- EP429_R1_trimmed.fastq 
- EP429_R2_trimmed.fastq
- EP429_R1un_trimmed.fastq 
- EP429_R2un_trimmed.fastq

___

### **2. Look at trimmed reads**
#### Forward Reads
```{bash, eval=FALSE}
cat Taxa/EP429/EP429_R1_trimmed.fastq | awk '((NR-2)%4==0){read=$1;total++;count[read]++}END{for(read in count){if(!max||count[read]>max) {max=count[read];maxRead=read};if(count[read]==1){unique++}};print total,unique,unique*100/total,maxRead,count[maxRead],count[maxRead]*100/total}'
```
##### Total number of reads= 224,973 
##### Total number unique reads= 78,527
##### Percentage of unique reads= 34.9051
##### Most abundant sequence= ATATGTACACCTTGAAGAAAAATATGTTTTTGGTATGTATGTCTAGGATATTATTAGGCCATTTCA
##### TTCATTTTTCTGGTGACTTTTCGCTGCTGGAGAATGCAAAATAATTTTGTAGACATGTCTAGGAAATTATTTTTGGTATGAATA
##### Frequency of most abundant sequence= 370
##### Percentage of total in file= 0.164464
___
### **3. Metagenome Assembly Using SPAdes**

```{bash, eval=FALSE}
spades.py --meta --pe1-1 ./Taxa/EP429/EP429_R1_trimmed.fastq --pe1-2 ./Taxa/EP429/EP429_R2_trimmed.fastq -o spades_output
```
#### This created an output file called spades_output.  Within this file we are specifically interested in one: scaffolds.fasta.  Because I am running 10 taxa, each scaffolds.fasta was renamed "EP*number*_scaffolds.fasta
___
### **4. Contig Binning using MetaBAT**
#### In order to start contig binnig, scaffolds.fasta had to be converted into a .bam file index.  I used the package bwa and samtools for this.
- Create reference index (bwa index is the command, -p reference is the output name for future commands, -a bwtsw is used because it is a large dataset, ref.fa is the input)
#### Build reference index for bowtie2
```{bash, eval=FALSE}
bowtie2-build -f ref.fa dbname
```
##### Reference genome used was Ephemera danica from NCBI
#### Go back and use the starting fastq files
```{bash, eval=FALSE}
bowtie2 -x dbname -1 EP429_R1_001.fastq -2 EP429_R2_001.fastq  --no-unal -S EP429.sam
```
##### Each taxa only had .5% to 4% mayfly DNA, must have been sloppy lab work
#### Convert .sam file to .bam file
```{bash, eval=FALSE}
samtools view -Sb EP429.sam > EP429.bam
```
#### Sort the .bam files
```{bash, eval=FALSE}
samtools sort EP429.bam -o EP429_sorted.bam
```
#### Contig binning using either metaBAT
```{bash, eval=FALSE}
runMetaBat.sh seqs.fa EP023_sorted.bam EP025_sorted.bam EP080_sorted.bam EP084_sorted.bam EP109_sorted.bam EP150_sorted.bam EP411_sorted.bam EP416_sorted.bam EP418_sorted.bam EP429_sorted.bam
```

##### Contig binning command isn't working, so I am currently at this step troubleshooting. 

___
### **5. Isolate Mayfly data**
```{bash, eval=FALSE}
```

___
### **6. Align Sequences using MUSCLE**
#### awk command merges all taxa .fasta files into one .fasta file, muscle command aligns multiple sequences using that file.
```{bash, eval=FALSE}
awk '/>/{sub(">", "&"FILENAME"_");sub(/\.fasta/,x)}1' *.fasta > seqs.fa
muscle -in seqs.fa -out seqs.afa -maxiters 1 -diags1 -sv
```

___
### **7. Phylogenetic Trees using IQTree**
```{bash, eval=FALSE}
```
